tests/syscall/fs/host/host.c

```diff
@@ -78,26 +89,105 @@ int wmain(int argc, wchar_t* argv[])
     // Windows does not support umask.
     // Please set up the right permission to the parent directory.
 
-    int ret = test_fs_posix(enclave_path, src_dir, tmp_dir);
+    rmdir(argv[3]);
 
-    free(src_dir);
-    free(tmp_dir);
+    r = oe_create_fs_enclave(enclave_path, type, flags, NULL, 0, &enclave);
+    OE_TEST(r == OE_OK);
 
-    return ret;
+    r = test_fs(enclave, src_dir, tmp_dir);
+    OE_TEST(r == OE_OK);
+
+    r = oe_terminate_enclave(enclave);
+    OE_TEST(r == OE_OK);
+
+    printf("=== passed all tests (hostfs)\n");
+
+    return 0;
 }
+#else
+int rmdir(const char* path)
+{
+    char abs_path[PATH_MAX];
+    int result = -1;
+    struct dirent* entry = NULL;
+    DIR* dir = NULL;
+    if (!(dir = opendir(path)))
+    {
+        goto done;
+    }
+
+    while ((entry = readdir(dir)))
+    {
+        DIR* sub_dir = NULL;
+        FILE* file = NULL;
+        memset(abs_path, 0, PATH_MAX);
+        if (!strcmp(entry->d_name, ".") || !strcmp(entry->d_name, ".."))
+        {
+            continue;
+        }
+
+        sprintf(abs_path, "%s/%s", path, entry->d_name);
+        if ((sub_dir = opendir(abs_path)))
+        {
+            closedir(sub_dir);
+            rmdir(abs_path);
+        }
+        else if ((file = fopen(abs_path, "r")))
+        {
+            fclose(file);
+            remove(abs_path);
+        }
+    }
+
+    result = remove(path);
```

Comment:
Are we missing a `closedir()` for the initial open of the root `path` before removing it?

Reply:
Fixed with nftw.